AGS02MA.h

```diff
@@ -28,6 +28,12 @@
 class AGS02MA
 {
 public:
+  struct RegisterData
+  {
+    uint8_t data[4];
+    uint8_t crc;
```

Comment:
Why copy the CRC in the register data?
If the CRC is not OK there is no data copied to the struct, so only correct CRC will be copied so it loses its value?

Reply:
Well, in line with what I commented in the readme:
> Primarily intended for troubleshooting and analysis of the sensor. Not recommended to build applications on top of this method's raw data.

I opted for the transparency of including it.
It might even be an idea to not require a valid CRC if we want the data 'even more raw', and move it to a bool,

```cpp
  struct RegisterData
  {
    uint8_t data[4];
    uint8_t crc;
    bool    crcValid;
  };
```

Reply:
I still think the first check (completion of the I2C command / TwoWire errors) before filling the struct. But the CRC would be optional. And IMO shouldn't return false from the method if CRC is bad, as that bool is in the struct.

It would simplify error handling, as you can use the struct when it returns true. Even if the CRC check failed, at least you can be sure the data came from this read, not random uninitialized data or buffers from previous reads.

Reply:
Example of still error checking, unplugging the sensor mid-read looks like this:

```
REG[0x0]        0       0       5       44      CRC: OK  91     Sensor data:    0       1324
REG[0x1]        0       125     5       129     CRC: OK  249    Calibration:    125     1409
REG[0x2]        0       100     1       247     CRC: OK  148    Calibration:    100     503
REG[0x3]        3       232     0       179     CRC: OK  171    Calibration:    1000    179
REG[0x4]        Error:  -12
REG[0x5]        Error:  -12
REG[0x6]        Error:  -12
REG[0x7]        Error:  -12
REG[0x8]        Error:  -12
REG[0x11]       Error:  -12
REG[0x20]       0       0       80      117     CRC: OK  61
REG[0x21]       26      229     26      229     CRC: OK  113    I2C address:    0x1A
```

Reply:
LGTM